docs/developer-guide/index.md

```diff
@@ -173,58 +173,53 @@ e2e-tests:
 
 ### NFD-Master
 
-When running as a standalone container labeling is expected to fail because
-Kubernetes API is not available. Thus, it is recommended to use `-no-publish`.
+For development and debugging it is possible to run nfd-master as a stand-alone
+binary outside the cluster. The `-no-publish` flag can be used to prevent
+nfd-master making changes to the nodes. If `-no-publish` is not set, nfd-master
+also requires the `NODE_NAME` environment variable to be set for cleaning up
+stale annotations.
```

Comment:
quick question, `KUBERNETES_NAMESPACE` isn't needed for the master?

Reply:
Good question. And correct, it is not needed for nfd-master (the namespace env is required for those daemons that create CRs). The nfd-master watches CRs in all namespaces